drivers/timer/mcux_os_timer.c

```diff
@@ -194,17 +222,23 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 		if (pm_state_next_get(0)->state == PM_STATE_STANDBY) {
 			uint64_t timeout;
 
-			/* Check the amount of time left and switch to a counter
-			 * that is active in this power mode.
-			 */
-			timeout = base->MATCH_L;
-			timeout |= (uint64_t)(base->MATCH_H) << 32;
-			timeout = OSTIMER_GrayToDecimal(timeout);
-			timeout -= OSTIMER_GetCurrentTimerValue(base);
-			/* Round up to the next tick boundary */
-			timeout += (CYC_PER_TICK - 1);
-			/* Convert to microseconds and round up to the next value */
-			timeout = (((timeout / CYC_PER_TICK) * CYC_PER_TICK) * CYC_PER_US);
+			if (wait_forever) {
+				timeout = UINT32_MAX;
+			} else if (counter_overflow) {
+				timeout = counter_remaining_ticks;
```

Comment:
shouldn't this unit be wrong ? this timeout value gets converted again to ticks, I think it will be an incorrect calculation